### DISCUSSION TITLE
如何解决CloudBeaver跳转不正确的问题

### DISCUSSION THREAD

**Discussion by Gygkimo95** [actiontech/sqle#1521] [Q&A] [answered]
现状：
* 已配置CloudBeaver工作台，使用8978端口登录CloudBeaver正常
* 使用SQLE跳转SQL工作台，无法正常跳转

**Comment by Gygkimo95** [comment]
当您配置CloudBeaver后，点击SQL工作台，跳转不正确时，您可按照下方的排查方法初步排除问题：

## 1. 确认CloudBeaver版本

SQLE目前支持的CloudBeaver版本为[22.2.0、22.2.1、22.2.3、22.3.1]（ 请参考：https://actiontech.github.io/sqle-docs/docs/user-manual/sql-workbench/how-to-configure ）
若您使用的CloudBeaver版本不在此范围内，将有可能造成跳转不正确的问题，建议将版本切换为当前已确认支持的版本。

## 2. 确认是否曾多次配置CloudBeaver

若您在当前环境下曾多次配置CloudBeaver，SQLE会记录先前的CloudBeaver配置缓存，导致当前配置下工作台跳转始终无法正常执行。
此时，需要您执行以下操作：
* 进入sqle业务库
  ![image](https://github.com/actiontech/sqle/assets/113898890/e4889d78-b1c1-4e46-a300-f5e668ae064f) 
* 清除CloudBeaver相关的缓存表的记录；
 
```sql
TRUNCATE TABLE `cloud_beaver_instance_caches`;
TRUNCATE TABLE `cloud_beaver_user_caches`;
```
  ![image](https://github.com/actiontech/sqle/assets/113898890/2d24c7b3-3191-4e8f-bec0-a1118ea5417a)
* 重启SQLE
  * 若未rpm方式安装，使用命令 systemctl restart sqled；
  * 若使用docker方式安装，使用命令docker restart sqle；

## 3. 清除页面缓存或使用浏览器无痕模式打开页面

若您已核对CloudBeaver版本无误、已清除数据库缓存表，并重启SQLE，此时仍跳转不正确时，可考虑清除浏览器缓存或使用浏览器无痕模式，重新打开SQLE；

若上述排查方法仍不能解决您的问题，请在社区群联系我们，我们将帮您解决CloudBeaver跳转失败问题。

**Comment by lumos0** [comment]
建议将此FAQ补充到文档中.

**Reply by Gygkimo95** [reply]
已在工作台配置方法页面，新增常见问题入口，感谢建议

**Comment by Seechi-Yolo** [comment, marked as answer]
# SQLE 3文档补充
当您配置CloudBeaver后，点击SQL工作台，跳转不正确时，您可按照下方的排查方法初步排除问题：

# 1. 确认CloudBeaver版本
SQLE目前支持的CloudBeaver版本为[22.2.0、22.2.1、22.2.3、22.3.1]（ 请参考：https://actiontech.github.io/sqle-docs/docs/user-manual/sql-workbench/how-to-configure ）
若您使用的CloudBeaver版本不在此范围内，将有可能造成跳转不正确的问题，建议将版本切换为当前已确认支持的版本。
# 2. 确认是否曾多次配置CloudBeaver
若您在当前环境下曾多次配置CloudBeaver，SQLE会记录先前的CloudBeaver配置缓存，导致当前配置下工作台跳转始终无法正常执行。
此时，需要您执行以下操作：

1. 进入dms业务库，清除CloudBeaver相关的缓存表的记录，执行：

```
use dms;
TRUNCATE TABLE `cloudbeaver_connection_caches`;
TRUNCATE TABLE `cloudbeaver_user_caches`;
```
![image](https://github.com/actiontech/sqle/assets/31072467/cbe04fa4-d81f-44a1-8741-bba8c4185e46)
![image](https://github.com/actiontech/sqle/assets/31072467/e9083278-d8a8-4102-9c61-1914a21b094f)

2. 重启SQLE
- 若使用rpm方式安装，请使用命令：
```
systemctl restart dms；
systemctl restart sqled；
```
- 若使用docker方式安装，请使用命令：`docker restart sqle；`
# 3. 清除页面缓存或使用浏览器无痕模式打开页面
若您已核对CloudBeaver版本无误、已清除数据库缓存表，并重启SQLE，此时仍跳转不正确时，可考虑清除浏览器缓存或使用浏览器无痕模式，重新打开SQLE；

若上述排查方法仍不能解决您的问题，请在社区群联系我们，我们将帮您解决CloudBeaver跳转失败问题。
